### PR TITLE
eachAsync: use nextTick instead of setTimeout and other small improvements

### PR DESCRIPTION
**Pull request by Uzlopak**

### CODE REVIEW

File: lib/helpers/cursor/eachAsync.js [hunk 1]
```diff
@@ -23,22 +23,20 @@ const promiseOrCallback = require('../promiseOrCallback');
 
 module.exports = function eachAsync(next, fn, options, callback) {
   const parallel = options.parallel || 1;
-  const batchSize = options.batchSize;
+  const batchSize = options.batchSize || undefined;
```
Comment (vkarpov15) [inline lib/helpers/cursor/eachAsync.js:26]: Why is this necessary? For one thing, this will cause `batchSize: 0` to throw an unreadable error.

File: lib/helpers/cursor/eachAsync.js [hunk 2]
```diff
@@ -23,22 +23,20 @@ const promiseOrCallback = require('../promiseOrCallback');
...
-      if (typeof batchSize !== 'number') {
-        throw new TypeError('batchSize must be a number');
-      }
-      if (batchSize < 1) {
-        throw new TypeError('batchSize must be at least 1');
-      }
-      if (batchSize !== Math.floor(batchSize)) {
-        throw new TypeError('batchSize must be a positive integer');
-      }
+  if (typeof batchSize !== 'undefined') {
```
Comment (vkarpov15) [inline lib/helpers/cursor/eachAsync.js:29]: Why move this out of `promiseOrCallback()`? We prefer to consistently return a promise, rather than sometimes return promise vs sometimes throwing a sync error - more like `async function()` semantics

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

File: lib/helpers/cursor/eachAsync.js (hunk 2 above)
Comment (Uzlopak) [reply lib/helpers/cursor/eachAsync.js:29]: Oh nice. Didn't thought about this.

Comment (Uzlopak) [reply lib/helpers/cursor/eachAsync.js]: fixed

File: lib/helpers/cursor/eachAsync.js (hunk 1 above)
Comment (Uzlopak) [reply lib/helpers/cursor/eachAsync.js]: Correct.

**Comment by Uzlopak** [comment]
@vkarpov15 
It should now be according to your liking :)

**Review by vkarpov15 — APPROVED** [review]
Thanks :+1:
